orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -1040,8 +1040,17 @@ common::Status GenerateSubgraph(Graph& graph, const Node* start_node) {
     }
   }
 
-  // update the grah with only visited inputs and outputs
-  graph.SetInputs({visited_inputs.begin(), visited_inputs.end()});
+  // If the following line is uncommented, middle and last pipeline stages may
+  // have unresolved symbolic shapes. The reason is that some symbolic shapes
+  // are defined for the original inputs, if original inputs are removed, we
+  // loss the hit to resolve symbolic shapes. For example, if an original
+  // input's shape is [batch, sequence, 1024], that input should be provided as
+  // a feed to all pipeline stages. Otherwise, we don't know the actual values
+  // of "batch" and "sequence".
+  //
+  // graph.SetInputs({visited_inputs.begin(), visited_inputs.end()});
```

Comment:
> graph.SetInputs({visited_inputs.begin(), visited_inputs.end()}); [](start = 4, length = 65)

Will this affect other newly-added inputs as well? Noted that SplitGraph() adds new inputs for send/recv for each stage. Without this call, inputs of send/recv from other partitions will also be included.

Reply:
No. This list is always `[input1, input2, input3]`. Here is the input list of the second stage.



---
In reply to: [439082782](https://github.com/microsoft/onnxruntime/pull/4204#discussion_r439082782) [](ancestors = 439082782)

Reply:
I am not sure why input set in SplitGraph() are not showing up here. Looking at the graph, it seems all the inputs are now doubled as outputs. I don't think this is related to this change or affecting the pipeline execution though. Let's merge this and I'll take a look and track down the issue. 

---
In reply to: [439651343](https://github.com/microsoft/onnxruntime/pull/4204#discussion_r439651343) [](ancestors = 439651343,439082782)